### DISCUSSION TITLE
[Design/Discussion] Lavaland Adventuring Overhaul

### DISCUSSION THREAD

**Discussion by matttheficus** [ParadiseSS13/Paradise#20164] [Documents]
# Lavaland Adventuring Overhaul 

**ITEMIZATION PATHS** (_read below_): https://docs.google.com/spreadsheets/d/1sl7dZir5KFdolJM32SEq0hO7VmNU0Vu0vbCxqCSbeIY

**DISCORD DISCUSSION**: https://discord.com/channels/145533722026967040/145700319819464704/1063239326013599844

Currently, Lavaland is in this odd state of power-escalation, speedrunning for loot/tendrils and VERY solo-minded mechanics surrounded by a passive-mining system. There is mininmal reason to work together with your fellow Miners/Spleunkers while adventuring, and even DOWNSIDES to working together to fight Megafauna/Tendrils/etc. The idea behind this document is to change Lavaland from a "collect the shiny toys by myself" to a system that encourages players to work together instead of nearly strangling eachother when one player "rushes all the good stuff".

---

## Issues with the Current System
- There is no way to share kills of Megafauna/Fauna between KC and KPA users.
- There is no "reset" mechanic that players can activate when all of a certain Megafauna are dead (think BotW's Blood Moons).
- There is no "progressive" tree for KPA users other than going to Science for upgrades. Likewise, there is no way for ALL KC users to progress at the same rate as not all trophies can be acquired, leading to people "speedrunning".
- There is no defensive progression to go through other than going from `Miner Suit > Drake Armor > HECK Suit`.
- KC Trophies are all RNG-based to drop, rather than something you can farm via crafting from killing (mega)faunas.
- There are minimal QoL items on Lavaland's loot table that help on Lavaland, most are "antag/overpowered" items (`meat hook, eye of god, cursed heart, inferno, warp cubes, dark blessing, and cult items`) or are VERY old designs that are nothing more than shiny trinkets due to how outclassed they are by basic tools (`Paradox Bags, diamond pickaxe, KA modkits, lava boat, possessed sword, lifesteal crystal, and the immortality crystal`).

# Possible Fixes

### Issues with Progression

I personally break this into four issues:
- KPA Progression
- KC Progression
- Defensive Progression
- QoL Progression

#### Foundational Progression Changes

Given that there will now be "paths" to choose from for Lavaland, the "starting point" for miners will have to be reworked. To do this, the Mining Voucher will be reworked to give different "starting points". This will give you two sets of options:

- KPA or KC
- A Starting Gearset

Starting Gearsets Include:

**Collector's Kit** - 1x Bulky Explorer's Webbing (more slots) + 1x Harvesting Tool (replaces survival knife) + 1x MedPen
**Miner's Kit** - 1x Larger Starting Mining Satchel + 1x Advanced Ore Scanner + 1x MedPen
**Explorer's Kit** - 1x Adventurer's Suit + 1x Explorer's Webbing + 1x MedPen
**Medic's Kit** - Adv Trauma Kit + 2x MedPen + Fulton Kit/Beacon

Along with this, Crafting and Gathering from Fauna/Flora/locations will become the new way to "move through" the progression systems. This will be mostly interfaced via using the Survival Knife as your main "collection" tool for crafting items. Applying it to Flora/Fauna/tendrils will allow you to "harvest" them for items. This will be the basis for collecting materials to use for upgrading your Armor/Weapons/crafting QoL items.

#### KPA Progression

This one is rather simple. I do like the idea of this staying locked behind Science/RND. ~~I would personally want the modkits to be removed from the Mining Vendors and placed solely within RND~~ **[CHANGE]**: Basic KA Upgrades will be available in the vendor, but T2+ will require a trip to Science. This causes there to be a symbiotic relationship between miners playing safe early to ensure Science is given mats and items to ensure they can get their KPAs upgraded. 

I would also like to see a more "attachment" based upgrade system, kind of like the KC so that it steps away from the "percentage based" system that is currently in the game as well as moving it from a "spam click" gameplay loop to a semi-reloadable system (think the Adamant Rail gun from Hades, with infinite ammo but a "reload")

Examples can be found **[HERE](https://docs.google.com/spreadsheets/d/1sl7dZir5KFdolJM32SEq0hO7VmNU0Vu0vbCxqCSbeIY)** on the "KPA Upgrades" sheet.

#### KC Progression

For the KC, I would like to move from an RNG-based full-drop system to a "crafting" system. You are choosing to go melee, a semi-tribal method using technology to assist you. Some of the trophies are VERY neat, and are probably some of the most enjoyable gameplay moments ive had in SS13's minimal PvE environment. But, the fact you need to speedrun to the Blood Drunk Miner to have a chance to get their trophy is rather rough. Below I will list a few ideas to change how the KC goes through its upgrade path:

- KC Trophies are made through the crafting menu from "salvaged" parts of creatures and plants around Lavaland. You can "skin" any killed faunas with the KC for "rare" materials used only in Trophy creation(see below for how skinning becomes a QoL progression). You would then combine these items with other items you collect to make the different Trophies.
(i.e. > `Watcher Sinew x4 + Plant Fibres x2 + Watcher Wings (R) x2 == Watcher Trophy`)
This would allow KC and KPA users to work together and trade items theyve "farmed" to encourage teamwork (and hell, you could ask people to pay for 'em)
- KC Trophies should not be locked behind KC-only combat. We should be encouraging players to work together as teams to take down Megas (think melee vs ranged DPS in an MMO).
- KCs should be able to melee-parry, to reduce full dependence on the BDM Eye trophy.

Examples can be found **[HERE](https://docs.google.com/spreadsheets/d/1sl7dZir5KFdolJM32SEq0hO7VmNU0Vu0vbCxqCSbeIY)** on the "KC Upgrades" sheet.

#### Defensive Progresion

One of the sorriest and most overlooked parts of Lavaland I feel is crafting. You have a whole list of Tribal items that really only have a use to Ashwalkers. I would like to expand this system as the main "goal" of Lavaland. Your survival knife goes from an underused melee attachment to your KPA and becomes a "skinning tool" to collect items and drops from your kills on Lavaland to be used to craft progressively helpful armor. For example, the progression system could look like this:

Examples can be found **[HERE](https://docs.google.com/spreadsheets/d/1sl7dZir5KFdolJM32SEq0hO7VmNU0Vu0vbCxqCSbeIY)** on the "Defensive Tree" sheet.
This entire system would mirror the KC Trophy crafting system and have recipes you'd need to farm to get.

This might seem like a scary thing to balance, but if it incorperates Lavaland items, we can introduce a "brittle" system that "nerfs" Lavaland items on Station to be on-par with Station-found items.

#### QoL Progression
Things such as healing, ore gathering, scanner upgrades, explorer mesh storage slots, etc can all be feasibly upgradable using Lavaland as a focus instead of locking it behind tendril loot only.

Other QoL ideas are some sort of on-Base Ore Deposit that sends to the ORM along with a stationary Mining Drill that you can place on Lavaland for passive material gain.

Examples can be found **[HERE](https://docs.google.com/spreadsheets/d/1sl7dZir5KFdolJM32SEq0hO7VmNU0Vu0vbCxqCSbeIY)** on the "QoL and Useful Items" sheet.

### Issues with Tendril Loot

To start, I would like to COMPLETELY remove the "breaking a tendril opens a chasm" mechanic. Instead, you can "salvage/mine" materials needed for crafting from the tendril, causing it to become dormant for a certain amount of time. You can either wait it out, or reinvigorate it with some kind of "restart item" (probably something you can craft with LL plants and a legion core). This prevents full round removal AND gives a reason to have "nests" all over LL that dont TRULY go away. 

***[NOTE: I WILL EXPAND ON THIS, BUT THINKING AT WORK IS DISRUPTIVE]***

### Ash Walkers
With a full revamp of defensive/offensive/QoL items being more Lavaland-derived, this would allow for the Ashwalkers to be a good "testing ground" for dead players/observers to spawn in and try to "climb the tech tree" of progression.
- Instead of the "Base Armor" being an explorer suit, they can craft the usual "Bone Armor" as their base option.
- Spears could be crafted into a tribal version of a KC, using the power of the Necropolis to empower their hits/give special effects. This would also encourage Miners to trade/fight the Ash Walkers for a chance to use a different toy.

**[THIS ALSO NEEDS SOME MORE FLESHING OUT, AHHHHHHHHH IDEAS]**

**Comment by matttheficus** [comment]
Spreadsheet for theorycrafting Lavaland Items/Progression/upgrades added to top of discussion.

**Comment by matttheficus** [comment]
## UPDATE NUMBER ONE

- Gearsets added to replace Mining Vouchers
- Most Default KC Trophies fleshed out (minus Tumor ones)
- Armor Baseline and Tier Progression outlined
- KPA needs a LOT of number crunching.
-- Additionally, need to decide on Attachment-based or Tinker-based (item attachments vs sliders vs mix)
- KPA progression might be trivial until after #20000 
- Megafauna/fauna might need balance tweaking, but this can be done later

**Comment by matttheficus** [comment]
## UPDATE NUMBER TWO

- Basic Crafting Materials laid out
- KA locked into an "attachements" idea. Utility "underbarrel" to replace the mining laser
- KC Trophies mostly finished idea wise.
- Defensive Progression fleshed out a bit more
- QoL Items added.
- Starting Gearsets tweaked

**Comment by matttheficus** [comment]
## UPDATE THREE

- More QoL items added.
- KA's fleshed out with an outline. Will likely need work on these. Current Max Values added to sheet for reference.
- More Crafting Materials added

Discussion updated to reflect current changes.

**Comment by Sirryan2002** [comment]
Ok I've finally looked over this doc and have some feedback. I'll start with overall general input and then a few specific points.

From what I'm gathering, you want lavaland to be based around the "adventure." So fighting monsters, working together to kill megafauna, and using the materials found in the world around you to progress yourself. You consider the mining part of lavaland to be a largely separate and disconnected part of the fighting aspect. While there are mining points involved where you trade ores mined for progression through vendors and R&D, you consider the actual mining part to be very passive and unrewarding

**General Feedback**

- I really like the idea of delving heavily into a crafting system as one of the main game loops of lavaland, if you can make just the act of crafting to be fun, I think it will contribute significantly towards a more rewarding miner round. So I'm curious as to how you specifically want to design the crafting system (the current crafting menu, combining items with each other, using some form of crafting table?). I also want to know how exactly you plan on distributing these items into lavaland and how you plan on making gathering stuff from flora actually interesting. With your proposed changes, killing megafauna and unlocking/hitting mid-to-high-level progression nodes is not what players will be doing minute by minute on lavaland, they will be collecting resources, crafting, and fighting low level mobs and if that's boring all the end stuff doesn't really matter.
- Why bother forcing players to go through R&D for upgrades? There is always the argument of inter-departmental reliance, but to be real, mining minerals is boring as all hell. Furthermore, players will need to go back upstairs to the station and walk themselves all the way over to R&D only to wait in lines or get told no. This will take the player out of the fun of lavaland for anywhere from 4 to 15 minutes. This is an important distinction to make because you want the progression to all be based on lavaland but you are also requiring the player to go back on station and hook into a completely different set of (unfun) features and loops to progress. Perhaps instead of making it so that KPA or Crusher upgrades are 100%  `item -> click on` based, they can be integrated into some kind of upgrade lab in science which is directly hooked into R&D and doesn't require scientists to print stuff. (this could be considered at a later time, but why is mining so god damn far away from R&D on maps?)
- This is an adventuring design doc but there is no mention of generic lavaland fauna. There are three types of regular mobs on lavaland and two of them are just basically reskins of each other (watchers & goliaths), Miners will be fighting these throughout your progression model in order to upgrade and get better, atm they're very dry to fight, legions in a group are by far the most interesting. I have a few questions about these, 1) how do you plan on changing them  2) do you plan on adding or removing regular fauna mobs? 3) how do you plan to distribute them on the map in relation to the mining outpost and ash walker encampment?
- Finally, how do you plan on keeping true to mining? If you're going to keep the reliance on R&D, players will have to mine not only as a requirement by SOP and also keeping the station fueled with resources but they need it to progress through this system. I think you should put some thought into how mining and adventuring can be much more closely tied together. Perhaps certain mobs can cluster around certain ore pockets or maybe make it so there are caverns full of high-value ore which are home to waves of mobs. Hell, you could even change out the randomly placed ores on the map for clusters instead.  
- You've come up with a very content-heavy system, i.e. you're going to need to add in a lot of new items. I recommend cutting down on stuff initially and keeping it in your back pocket. I would really like to see a lot of thought put into how each base weapon or armor works mechanically and making that really good before considering adding variants to it.

**Specific Feedback**

- I noticed with the roundstart gearsets you were going for a class system. Why is the harvester tool only limited to one class, can you buy it in vendors? It seems like a really important tool to what you're proposing, this class may become a dominant pick.
- I really like the cloaks and small shelters, mining pods are ugly bulky beasts that aren't needed when you need a pit stop because of a very inconvenient ash storm. This will be a better alternative
- The boot sheath should probably be usable like a holster or PDA pen, otherwise it just turned into something like a jacket pocket and won't be used ever


When I think of Lavaland, and particularly this type of system, I imagine Deep Rock Galactic. It's a really great example of a combat-mining adventure slapped into one and could be a really good source of inspiration. I know I said a lot of things to keep in mind and I do hope that you ponder some of the questions presented because, in the end, the way you design this system matters a lot more than the content you add.

**Comment by matttheficus** [comment]
Thanks for the look over! Ill address some of the main points section-by-section.

## General Feedback

- Currently Crafting is rather scuffed in game. I'm unsure if I'll want to piggyback off the current system or make a "hub" device to work off of (a crafting bench, though more Rimworld and less Minecraft). In regards to the act of collecting items being "boring", that is some part of it. It IS busywork but its also downtime between "Bosses" so that people can turn in materials, get some medical supplies, and replace the monotonous "farm Legion Tendrils for 10 minutes" loop we currently have. An active action beats standing at a tendril for your entire "gear up" time.
In terms of distribution, the idea is currently to have a few "main" sources of your base materials:
**Tier One**: plants, goliaths, watchers, and legions.
**Tier Two**: crafted components, tendrils, "shiny" creatures (I'll be adding these to Tendril Spawns as a "rare mob").
**Tier Three**: Early-Game Megas (Drake, Heiro, BDM)
**Tier Four**: Unique Megafauna
Tier One and Tier Two are things you'll passively get while running about Lavaland, and as long as you make an effort to stop and collect them, you wont really be falling behind at all.
Tier Three and Tier Four are all things you aim for killing anyway, so if your plan is to go adventuring you'll likely be dealing with them anyway.
- In terms of avoiding the dreaded RND run, it is a necessary evil, at least with how RND is currently designed. That, and early tier KA upgrades will still be available in the Mining Vendor, but if you want maximum upgrades, you will have to go to RND. If you want to go the "safer" ranged option, you'll need to interface with the tech-savvy Science nerds. Perhaps Science can get some sort of "vendor upgrade" that makes it so that miners can then just get their upgrades from the vendors, but that unfortunate loop will likely be here until after #20000. I did think about making a "tinker" system instead, so maybe this can be built into the "Crafting Hub" as mentioned in the previous point.
- Tweaks and additions to Lavaland creatures and placements are secondary in my opinion. A base system at the player-level is necessary first and then more creatures can be added. At the moment, the ideal design phases goes something like this, with each Phase being its own PR:

**Phase One: KC and KA Rework** - this will simply lay foundational work for the KC's new trophy system by replacing current trophy drops with the "final crafted versions" as well as add the Blessing system WITHOUT crafting being implemented yet. This will also have the KA rework, with the upgrades being simple Vendor/Science items. KCs and KAs have a lot of balance implications for Crew AND Lavaland, so this should be dealt with first.
**Phase Two: Crafting, Defensives and QoL** - this will be where crafting is fully fleshed out and added, changing the Trophy system to a craft-based system, adding some QoL items, and implementing the Armor path.
**Phase Three: Fauna Tuning and Drop Rate Tuning** - this will be where drop amounts, damage, and other features can be tuned to prevent someone from maxing out or speedrunning the whole system in 20 minutes. Currently, the aim is to have you capable in a TEAM of clearing Lavaland's Unique Megafauna by the 1:45 mark for the round. This will take some tweaking and testing to get right.

- Base Lavaland changes would actually be wonderful. Currently, mining itself is very much "get plasmacutter, go in a straight line". I would love to touch on adding more features to Lavaland directly, but that approaches a Project-level undertaking and would put a LOT on my plate alone. This only looks to add a base system to the Adventuring side of things while keeping mining as a passive action you take while you scour Lavaland for materials and such. I would love to follow this up with a Mining Overhaul, but I have immensely less ideas for it than the Adventuring side at this point in time.
- Content-wise, the area I worry about becoming "bloated" is QoL. I might spin that into its own set of ideas to focus on AFTER Weapons and Armor are fleshed out so that they can get the proper amount of thought behind them. I'm working with Qwerty some on thinking through different ideas for unique end-game upgrades/items (the KA's Vetus mod, for example).

## Specific Feedback
- The Class system is most of the inspiration. The Harvester's Tool will only give the amount of bonus materials needed to help a team of 3-4 fully gear up per kill, rather than 2-3. It WILL likely be picked a lot, but more so in a team setting.
- The Cloak idea is something I've toyed with on its own for a few months, and honestly, I think it makes for great flavor (and fashion!)
- It will probably be a ctrl + click on the boot, like a PDA.
